setup.py

```diff
@@ -99,6 +100,30 @@ def run_tests(self):
         sys.exit(errno)
 
 
+class ConditionalCFFIBuildExt(build_ext):
+
+    def pre_run(self, ext, ffi):
+        from cffi import FFI
+        count = 0
+        for conditional in getattr(ffi, "_conditional_cdefs", []):
+            for attribute, (tc, rc) in conditional.items():
+                modname = "_cryptography_cffi_test_{0}".format(count)
+                tffi = FFI()
+                tffi.cdef("static const int {0};".format(attribute))
+                tffi.set_source(
+                    modname, tc,
+                    libraries=self.get_libraries(ext),
+                )
+                tmpdir = os.path.join(self.build_temp, "conditionals")
+                tffi.compile(tmpdir=tmpdir)
+                sys_path = sys.path[:]
+                sys.path.append(tmpdir)
+                mod = __import__(modname)
+                sys.path = sys_path
+                if getattr(mod.lib, attribute, None):
```

Comment:
I just have a habit of explicitly adding the ``None`` to ``getattr`` (even though it's not needed) to be explicit about the default (and for symmetry with ``{}.get(..., None)``),